### PR TITLE
feat(program): encodable program

### PR DESCRIPTION
**Pull request by 0xLucqs**
# TITLE

## Description

Description of the pull request changes and motivation.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [ ] CHANGELOG has been updated.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -60,6 +60,8 @@ felt = { package = "cairo-felt", path = "./felt", default-features = false, feat
 ] }
 bitvec = { version = "1", default-features = false, features = ["alloc"] }
 
+parity-scale-codec = { version = "3.6.0", no-default-features = true }
```
Comment (AbdelStark) [inline Cargo.toml:63]: it's not the same version we use in Madara, can we try to uniform and use same versions. Fine to upgrade Madara to this version if it works

File: felt/src/bigint_felt.rs [hunk 2]
```diff
@@ -35,6 +36,82 @@ lazy_static! {
 pub(crate) struct FeltBigInt<const PH: u128, const PL: u128> {
     val: BigUint,
 }
+#[cfg(feature = "scale-codec")]
+impl<const PH: u128, const PL: u128> Encode for FeltBigInt<PH, PL> {
+    fn encode_to<T: Output + ?Sized>(&self, dest: &mut T) {
```
Comment (AbdelStark) [inline felt/src/bigint_felt.rs:41]: well commented 👏

File: vm/src/lib.rs [hunk 3]
```diff
@@ -7,7 +7,6 @@
 
 #![cfg_attr(docsrs, feature(doc_cfg))]
 #![deny(warnings)]
-#![forbid(unsafe_code)]
```
Comment (AbdelStark) [inline vm/src/lib.rs:10]: Would be great to add a comment somewhere to explain why we allow unsafe code. also, is there a way to allow it only on places we need and not at the crate level ?

**Review by AbdelStark — APPROVED** [review]
left some comments, non blocking ones though

File: Cargo.toml (hunk 1 above)
Comment (0xLucqs) [reply Cargo.toml:63]: cargo handles it automatically but yes we should update the madara version

**Pull request merged**